### PR TITLE
added test cases for vector class

### PR DESCRIPTION
**Pull request by helblazer811**

### CODE REVIEW

**Comment by helblazer811** [comment]
I can still test more edge cases.

**Comment by smoitra87** [comment]
Please remove compilation files such as .class and .ctxt using .gitignore

File: Vector/Vector.ctxt [hunk 1]
```diff
@@ -3,24 +3,22 @@ comment0.params=
...
-comment14.target=boolean\ isEmpty()
-comment15.params=index
-comment15.target=void\ remove(int)
-comment16.params=index\ element
-comment16.target=void\ set(int,\ java.lang.Object)
-comment17.params=index
-comment17.target=java.lang.Object\ get(int)
-comment18.params=
-comment18.target=java.lang.String\ toString()
+comment10.params=a
```
Comment (smoitra87) [inline Vector/Vector.ctxt]: This while file should not be part of the commit. Please set up .gitignore

File: Vector/Vector.java [hunk 2]
```diff
@@ -53,6 +53,7 @@ public void insertionSort(){
     }
     
     public void selectionSort(){
+
```
Comment (smoitra87) [inline Vector/Vector.java]: extra linefeed. Please remove.

File: Vector/VectorTest.java [hunk 3]
```diff
@@ -1,22 +1,155 @@
...
+ import static org.junit.Assert.*;
+import org.junit.Test;
 public class VectorTest
 {
-  public static void main(String [] args){
-      Vector vec=new Vector();
+    private Vector vector;
+      private void setUpVector(){
+           Vector vec=new Vector();
+           vec.add(10);
```
Comment (smoitra87) [inline Vector/VectorTest.java]: All these vec.add statements can belong to a for loop reading from an array.

Comment (smoitra87) [inline Vector/VectorTest.java]: Same for similar situations below.

**Comment by helblazer811** [comment]
I changed the things you commented on. I also think i want to start a symbolic computation project. It will involve creating Abstract Syntax Trees based on mathematical functions and doing things with them like simplification, factorization etc.
